pkg/ttnpb/identifiers.go

```diff
@@ -240,3 +240,34 @@ func (ids *UserIdentifiers) ValidateContext(context.Context) error {
 	}
 	return nil
 }
+
+// ExtractRequestFields lets gRPC middleware extract fields from request messages that embed these identifiers.
+func (ids *ApplicationIdentifiers) ExtractRequestFields(m map[string]interface{}) {
```

Comment:
Any reason not to have these receivers by value ?

Reply:
Yes. Which reminds me: https://github.com/TheThingsNetwork/lorawan-stack/pull/3535/commits/5ef1024a6e8ebe1c021248132f4af7b20d696da2